src/quo2/components/inputs/search_input/view.cljs

```diff
@@ -0,0 +1,88 @@
+(ns quo2.components.inputs.search-input.view
+  (:require [oops.core :as oops]
+            [quo2.components.icon :as icon]
+            [quo2.components.inputs.search-input.style :as style]
+            [react-native.core :as rn]
+            [reagent.core :as reagent]))
+
+(def ^:private tag-separator [rn/view {:style style/tag-separator}])
+
+(defn- inner-tags
+  [tags-coll]
+  (into [rn/view {:style style/tag-container}]
+        (interpose tag-separator)
+        tags-coll))
+
+(defn- clear-button
+  [{:keys [on-press blur? override-theme]}]
+  [rn/touchable-opacity
+   {:style    style/clear-icon-container
+    :on-press on-press}
+   [icon/icon :i/clear
+    {:color (style/clear-icon blur? override-theme)
+     :size  20}]])
+
+(defn handle-backspace
+  [^js/Object event scroll-view-ref]
+  (when (= (oops/oget event "nativeEvent.key") "Backspace")
+    (.scrollToEnd scroll-view-ref #js {:animated false})))
+
+(def ^:private props-to-remove
+  [:cursor-color :placeholder-text-color :editable :on-change-text :on-focus
+   :on-blur :on-clear :value :tags :disabled? :blur? :customization-color :override-theme])
+
+(defn- add-children
```

Comment:
Oh, that `add-children` function is not a reagent component.

It's a helper function that receives a component and a seq of children (already passed as varargs) and adds it as component's children if they aren't empty.

I created this function because it was cleaner than directly adding them in the render function.